Small patches

Pululut opened the pull request:
Fixes quite a lot of the things I mentioned in the thread: https://forums.transbian.love/?page=post&id=1063

Happy to help out. Please test and audit my changes before actually submitting them, I haven't tested but they should be fine, I've done much of this before lol.

Pululut committed:
Fix breadcrumbs when AJAX is enabled.

Pululut committed:
Force utf8mb4 collation for new data

Also changed varchar sizes and added $hugeInt here for the sakes of consistency so install doesn't break with "maximum key length is 767 bytes"

Pululut committed:
inconsistent text var fixed

also var128 didn't need updating

Pululut committed:
A bit more consistency in the schema

Prevents big useragents overfilling the table after utf8mb4 patches, too.

Pululut committed:
E_NOTICE: No way, I can't believe this!

real description: that one isset() will help reduce the amount of undefined variable errors everywhere significantly, lol

Pululut committed:
doBoardVersionFooter() is now a thing :P

Pululut committed:
Footer version now determined from version.php

Pululut committed:
A nicer, more detailed README in the works.

Pululut committed:
Fix getquote checking for wrong thing.

Pululut committed:
remove birthday code from wrong place

Pululut committed:
fix the issecure() error to actually be an error

Pululut committed:
wait why are there two functions for the same thing

Pululut committed:
Move code to the right place...

Pululut committed:
Secure search and make it publicly accessible.

Lana-chan commented on installSchema.php:
```diff
@@ -8,8 +8,8 @@
 $var128 = "varchar(128)".$notNull;
-$var256 = "varchar(256)".$notNull;
-$var1024 = "varchar(1024)".$notNull;
+$var256 = "varchar(191)".$notNull;
+$var1024 = "varchar(767)".$notNull;
```
what's the reason behind making these smaller?

Lana-chan commented on lib/mysqlfunctions.php:
```diff
@@ -1,12 +1,14 @@
 $var128 = "varchar(128)".$notNull;
-$var256 = "varchar(256)".$notNull;
-$var1024 = "varchar(1024)".$notNull;
+$var256 = "varchar(191)".$notNull;
+$var1024 = "varchar(767)".$notNull;
```
same comment as previous here

Lana-chan commented on pages/search.php:
```diff
@@ -37,8 +37,6 @@
 	</script>
 ";
 
-if($loguser['powerlevel'] >= 1)
-{
```
this would open search for guests, correct? if so, that would be undesired

Lana-chan commented on pages/search.php:
```diff
@@ -74,14 +72,9 @@
 
 echo "</td></tr></table>";
 
-
-if($loguser['powerlevel'] < 1)
-	throw new KillException();
-
```
same comment as previous

Lana-chan commented on README.md:
````diff
@@ -1,27 +1,40 @@
-## Warning
+## Warning!!
 
 Please note that you are using a version of Lesbiaboard obtained from the Git
-repository. This version of Lesbiaboard may have serious vulnerabilites and
-might not work at all.
+repository. It is recommended that you use a stable version from the Releases tab of the repository,
+rather than an interim version (unless you are testing for us, which we are more than happy to see).
 
+```"If any of you, in any way, hate homosexuals, people of a different color or women, please do this one favor for us — leave us the fuck alone."```
+
+## Requirements and Recommendations
+
+* PHP 7.3 or newer.
+* MariaDB 10.1 or newer. Oracle MySQL >5.7 is not supported.
+* HTTP web server
+* PHP extensions: mysqli, gd, xml, curl (for Discord Integration)
+ 
 ## Installation
 
 [Download Lesbiaboard](https://github.com/transbian-alliance/lesbiaboard/archive/master.zip),
-put it on a HTTP server supporting PHP and visit the location of Lesbiaboard
+upload it to the web root of your server (usually /www/ or /public_html/), and visit the location of Lesbiaboard
 in your server. The interactive installer should take care of everything.
 
 For more information, check the documentation [here](https://github.com/transbian-alliance/lesbiaboard/wiki/Installing).
 
 ## Feedback
 
-Create an issue [here on this repository](https://github.com/transbian-alliance/lesbiaboard/issues) for any bugs and requests.
+Create an issue [here on this repository](https://github.com/transbian-alliance/lesbiaboard/issues) for any bug reports.
+General support is available on [the Lesbiaboard forum](https://forums.transbian.love), too.
 
 ## Contributions
 
 Fork the repository and send a pull request. This project uses tab-based
 indentation and opening braces on separate lines, so keep that in mind while
 contributing patches.
+
+You can also contribute in other ways too, such as documentation, translations, being an active member of the forums or spreading the word if you like our software, creating themes, ranksets and/or plugins, or hell, if you feel really nice, making a toasted sandwich breakfast for us while we're snuggled up in bed one morning... <3
````
please do not modify the readme

Pululut committed:
as requested by maple

Pululut committed:
revert my readme changes, sorry X(

Pululut commented:
reverted README and added killswitch for guests on search page. the varchar is reduced because of utf8mb4 on a fresh install - despite enabling support you'll want to convert all the tables manually on your own end too so it works correctly, however there is a bug in some mysql setups where the key length cannot exceed a certain length and it will bugger up on install - i believe mysql 5.7 and some versions of mariadb do this, as well as when the engine is InnoDB (though it's forced to myisam currently anyways)
